proto/spacemesh/types.proto

```diff
@@ -24,14 +28,21 @@ message Account {
     Amount balance = 3;     // known account balance
 }
 
-message Reward {
-    uint64 layer = 1;
-    Amount total = 2;
-    Amount layer_reward = 3;
-    uint64 layer_computed = 4; // layer number of the layer when reward was computed
-    // tx_fee = total - layer_reward
-    AccountId coinbase = 5; // account awarded this reward
-    SmesherId smesher = 6;  // it will be nice to always have this in reward events
+message AccountIdWithOffset {
+    AccountId account_id = 1;
+    uint64 offset = 2;
```

Comment:
Doesn't having an offset imply persistent state in the backend? Can we get away with this without adding state/complexity to go-spacemesh?

Reply:
We said we want it to support pagination. I don't know how else to do pagination besides adding offset and optionally number of requested results to client requests. It will need to be evaluated against the servers notion of a set of matching items.

Reply:
Actually, I take this back. The state is in the API call, and doesn't necessarily need to be stored on the backend. The backend can just reload everything and respond from the offset.